python/eos/plot/plotter.py

```diff
@@ -283,8 +283,27 @@ def __init__(self, plotter, item):
         def plot(self):
             item = self.item
             oname = item['observable']
+            eos.Observables.assert_valid_name(oname)
             eos.info('   plotting EOS observable "{}"'.format(oname))
 
+            # handle plot variable, create kinematic or parameter
+            if not 'variable' in item:
+                raise ValueError("Missing key for plot of observable '" + oname + "': 'variable'")
+            try:
+                eos.Parameters.assert_valid_name(item['variable'])
+                var = parameters.declare(item['variable'], np.nan)
+            except ValueError:
+                eos.Observables.assert_valid_name_and_kinematic(oname, item['variable'])
```

Comment:
Why do you assert again the name of the observable? You did so in line 286 above.
All you want is the observable entry, which you can obtain in line 286 already with the proposed change for the first commit (return the observable entry for a valid name).

Reply:
Thanks, changed. Also, asserting the kinematic argument is not a responsibility of the `Observables` class, conceptually.